### PR TITLE
[DI] Added migration guide markdown

### PR DESCRIPTION
**Pull request by mssfang**

### CODE REVIEW

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+[DocumentIntelligenceAdministrationClientBuilder]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/src/main/java/com/azure/ai/documentintelligence/documentanalysis/administration/DocumentIntelligenceAdministrationClientBuilder.java
+[DocumentIntelligenceAdministrationClient]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/src/main/java/com/azure/ai/documentintelligence/documentanalysis/administration/DocumentIntelligenceAdministrationClient.java
+[DocumentIntelligenceAdministrationAsyncClient]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/src/main/java/com/azure/ai/documentintelligence/documentanalysis/administration/DocumentIntelligenceAdministrationAsyncClient.java
+[FormRecognizerMigrationInstantiateClient]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/formrecognizer/azure-ai-formrecognizer/migration-guide.md#instantiating-clients
```
Comment (YalinLi0312) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:189]: Looks this one is redundant, I don't see it is used in this file.

Comment (mssfang) [reply sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:189]: Yeah. You are right

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+## Breaking Changes
+
+### Clients names updates
+To improve the development experience and address the consistent feedback across the Document Intelligence SDK, this new
+version of the library introduces two new clients
+`DocumentIntelligenceClient` and the `DocumentIntelligenceAdministrationClient` that provide unified methods for
+analyzing documents and provide support for the new features added by the service in
+API version `2023-10-31-preview` and later.
+
```
Comment (YalinLi0312) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:101]: nit: Can you have a table to show the relationship of SDK and API version which would more straightforward?

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+          client.beginAnalyzeDocument("prebuilt-layout", null,
+                  null,
+                  null,
+                  Arrays.asList(DocumentAnalysisFeature.FORMULAS),
+                  null,
+                  null,
+                  new AnalyzeDocumentRequest().setBase64Source(Files.readAllBytes(document.toPath())));
+    ```
+
+  For the complete sample, see [Sample: KeyValuePair][Sample-AnalyzeAddOnKeyValuePair.
```
Comment (YalinLi0312) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:173]: missing the end `]`

**Comment by YalinLi0312** [comment]
Can you ref the migration guide in readme? I think most customers would check the main README first.

**Review by YalinLi0312 — APPROVED** [review]
LGTM!

**Pull request merged**